lib/commands/xctest.js

```diff
@@ -0,0 +1,240 @@
+ * Asserts that IDB is present and that launchWithIDB was used
+ *
+ * @param {object} opts Opts object from the driver instance
+ */
+export function assertIDB (opts) {
+  if (!opts.device?.idb || !opts.launchWithIDB) {
+    log.errorAndThrow(`To use XCTest runner, IDB (https://github.com/facebook/idb) must be installed ` +
+      `and sessions must be run with the "launchWithIDB" capability`);
+  }
+  return opts.device.idb;
+}
+
+/**
+ * Parse the stdout of XC test log
+ * @param {string} stdout A line of standard out from `idb xctest run ...`
+ * @returns {Array<Object>} results The final output of the XCTest run
+ */
+export function parseXCTestStdout (stdout) {
+  // Parses a 'key' into JSON format
+  function parseKey (name) {
+    const words = name.split(' ');
+    let out = '';
+    for (const word of words) {
+      out += word.substr(0, 1).toUpperCase() + word.substr(1);
+    }
+    return out.substr(0, 1).toLowerCase() + out.substr(1);
+  }
+
+  // Parses a 'value' into JSON format
+  function parseValue (value) {
+    value = value || '';
+    switch (value.toLowerCase()) {
+      case 'true': return true;
+      case 'false': return false;
+      case '': return null;
+      default: break;
+    }
+    if (!isNaN(value)) {
+      if (!_.isString(value)) {
+        return 0;
+      } else if (value.indexOf('.') > 0) {
+        return parseFloat(value);
+      }
+      return parseInt(value, 10);
+    }
+  }
+  if (!stdout) {
+    return [];
+  }
+
+  // Parse each line into an array
+  const lines = stdout.trim().split('\n');
+
+  // One single string, just return the string
+  if (lines.length === 1 && !lines[0].includes('|')) {
+    return [lines[0]];
+  }
+
+  const results = [];
+  for (const line of lines) {
+    // The properties are split up by pipes and each property
+    // has the format "Some Key : Some Value"
+    const properties = line.split('|');
+
+    // Parse each property
+    const output = {};
+    let entryIndex = 0;
+    for (const prop of properties) {
+      if (entryIndex === 0) {
+        // The first property only contains one string that contains
+        // the test name (e.g.: 'XCTesterAppUITests - XCTesterAppUITests.XCTesterAppUITests/testExample')
+        output.testName = prop.trim();
+      } else {
+
+        let [key, value] = prop.split(':');
+        output[parseKey(key.trim())] = parseValue(value ? value.trim() : '');
+      }
+      entryIndex++;
+    }
+    // Add this line to the results
+    results.push(output);
+  }
+  return results;
+}
+
+/**
+ * @typedef {Object} XCTestResult
+ *
+ * @property {string} testName Name of the test (e.g.: 'XCTesterAppUITests - XCTesterAppUITests.XCTesterAppUITests/testExample')
+ * @property {boolean} passed Did the tests pass?
+ * @property {boolean} crashed Did the tests crash?
+ * @property {number} duration How long did the tests take (in seconds)
+ * @property {string} failureMessage Failure message (if applicable)
+ * @property {number} location
```

Comment:
what is `location`?

Reply:
There was no documentation on it, so not sure.